Ginger/Unit Tests/NonUITests/WebServicesTest.cs

```diff
@@ -398,7 +399,50 @@ private string getXML()
 
         }
 
+        [TestMethod]
+        [Timeout(10000)]
+        public void WebServices_SoapWrapperActionTest()
+        {
+            WebServicesDriver mDriver = new WebServicesDriver(mBF);
+
+            Agent wsAgent = new Agent();
+            wsAgent.DriverType = Agent.eDriverType.WebServices;
+            wsAgent.Driver = mDriver;
+            ApplicationAgent mAG = new ApplicationAgent();
+            mAG.Agent = wsAgent;
 
+            mGR = new GingerRunner();
+            mGR.SolutionAgents = new ObservableList<IAgent>();
+            mGR.SolutionAgents.Add(wsAgent);
+
+            mGR.BusinessFlows.Add(mBF);
+
+            Activity Activity2 = new Activity();
+            Activity2.Active = true;
+            Activity2.ActivityName = "Soap UI Wrapper action";
+            Activity2.CurrentAgent = wsAgent;
+            mBF.Activities.Add(Activity2);
+
+            ActWebAPISoap actSoapUi = new ActWebAPISoap();
+
+            var xmlFilePath = TestResources.GetTestResourcesFile(@"XML\calculator_soapui_project.xml");
+
+            actSoapUi.AddOrUpdateInputParamValue(ActSoapUI.Fields.ImportFile, xmlFilePath);
+
+            mBF.Activities[0].Acts.Add(actSoapUi);
+
+            if (actSoapUi.ActInputValues.Count > 0)
```

Comment:
we usually do not use foreach inside unit test.  Also the assert is not clear, there is if condition to check contains and then you doing assert. So it will always be true.

We should do assert on something like , Inputvalues count or Properties count etc.